docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1021 @@
+----------------------------------
+{plugintype} {
+  {pluginname} {...}
+}
+----------------------------------
+endif::encode_method[]
+
+// /////////////////////////////////////////////////////////////////////////////
+// If encode_method IS defined (i.e. for codecs only)
+// /////////////////////////////////////////////////////////////////////////////
+ifdef::encode_method[]
+[source,js]
+[subs="attributes"]
+----------------------------------
+input {
+  codec => {pluginname} {...}
+}
+----------------------------------
+
+Or if using the codec in an output block:
+
+[source,js]
+[subs="attributes"]
+----------------------------------
+output {
+  codec => {pluginname} {...}
+}
+----------------------------------
+endif::encode_method[]
+
+==== Configuration Parameters
+[source,ruby]
+----------------------------------
+  config :variable_name, :validate => :variable_type, :default => "Default value", :required => boolean, :deprecated => boolean
+----------------------------------
+The configuration, or `config` section allows us to define as many (or as few)
+parameters as are needed to enable Logstash to process events.
+
+There are several configuration attributes:
+
+* `:validate` - allows developers to enforce a particular data type is passed to
+Logstash for this configuration option, e.g. `:string`, `:password`, `:boolean`,
+`:number`, `:array`, `:hash`, `:path` (i.e. a file-system path), `:codec`
+(since 1.2.0), `:bytes` (starting in 1.5.0).  Note that this also works as a
+coercion in that if I specify "true" for boolean (even though technically a
+string), it will become a valid boolean in the config.  This coercion works for
+the `:number` type as well where "1.2" becomes a float and "22" is an integer.
+* `:default` - lets you specify a default value for a parameter
+* `:required` - whether or not this parameter is mandatory (a Boolean `true` or
+`false`)
+* `:deprecated` - informational (also a Boolean `true` or `false`)
+
+=== Plugin Methods
+
+{methodheader}
+
+// /////////////////////////////////////////////////////////////////////////////
+// If register_method is defined (should be all types)
+// /////////////////////////////////////////////////////////////////////////////
+ifdef::register_method[]
+==== The `register` Method
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+  public
+  def register
+  end # def register
+----------------------------------
+
+The Logstash `register` method is like an `initialize` method. It was originally
+created to enforce having `super` called, preventing headaches for newbies.
+(Note: It may go away in favor of `initialize`, in conjunction with some
+enforced testing to ensure `super` is called.)
+
+`public` means the method can be called anywhere, not just within the class.
+This is the default behavior for methods in Ruby, but it is called explicitly
+here anyway.
+
+You may also assign instance variables here (i.e. variables prepended by `@`).
```

Comment:
May implies permission, which can lead to some interesting translations. In general, replacing "may" with "can" (appropriate in this case) or "might" is preferred.